### DISCUSSION TITLE
Modular architecture

### DISCUSSION THREAD

**Discussion by yablochnyi** [canyongbs/advisingapp#667] [Q&A] [answered]
Hello, I noticed you have a modular architecture with Filament. Could you tell me how you generate resources so that Filament understands in which module they need to be published and how to configure it so that Filament knows where the resources are located?

**Comment by dgoetzit** [comment, marked as answer]
Hello @yablochnyi,

In order to gain an understanding of how we use Filament with a modular approach, please take a look at the Filament plugin documentation: https://filamentphp.com/docs/3.x/panels/plugins

An example of our usage in the app can be seen through any one of the “plugins” per module, like https://github.com/canyongbs/advisingapp/blob/main/app-modules/student-data-model/src/StudentDataModelPlugin.php.

By registering these plugins with the default panel, we are then able to choose the resource namespace to place a new resource in when going through the prompt workflow of calling `php artisan make:filament-resource`
